python/erdos/streams.py

```diff
@@ -101,32 +146,48 @@ def internal_watermark_callback(coordinates, is_top):
 
 
 class WriteStream(object):
-    """Sends data and invokes callbacks when messages are received.
-
-    Handlese serialization of messages, and wraps an `internal.PyWriteStream`.
+    """ A :py:class:`WriteStream` allows an :py:class:`Operator` to send 
+    messages and watermarks to other operators that connect to the 
+    corresponding :py:class:`ReadStream`.
 
-    `_py_write_stream` is set during erdos.run(), and should never be set
-    manually.
+    Note:
+        `_py_write_stream` is set during erdos.run(), and should never be set
+        manually.
     """
-    def __init__(self, _py_write_stream=None):
+    def __init__(self, _py_write_stream=None, _name=None, _id=None):
         self._py_write_stream = PyWriteStream(
         ) if _py_write_stream is None else _py_write_stream
+        self._name = _name
+        self._id = _id
 
-    def is_closed(self):
+    @property
+    def name(self) -> Union[str, None]:
+        """ The name of the stream. (`None` if no name was given.) """
+        return self._name
+
+    def is_closed(self) -> bool:
         """Whether a top watermark message has been sent."""
         return self._py_write_stream.is_closed()
 
-    def send(self, msg):
+    def send(self, msg: Message):
         """Sends a message on the stream.
 
         Args:
-            msg (Message): the message to send. This may be a `Watermark` or a
-                `Message`.
+            msg: the message to send. This may be a `Watermark` or a `Message`.
```

Comment:
pyclass:`Watermark` and pyclass:`Message`? Might need to specify pyclass:`erdos.Message` if Sphinx isn't detecting it.